Sources/GRPCHTTP2TransportNIOPosix/GRPCHTTP2ServerTransportBootstrapper.swift

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import GRPCHTTP2Core
+import NIOCore
+import NIOHTTP2
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension ChannelPipeline.SynchronousOperations {
+  public typealias HTTP2ConnectionChannel = NIOAsyncChannel<HTTP2Frame, HTTP2Frame>
+  public typealias HTTP2StreamMultiplexer = NIOHTTP2Handler.AsyncStreamMultiplexer<
+    (NIOAsyncChannel<RPCRequestPart, RPCResponsePart>, EventLoopFuture<MethodDescriptor>)
+  >
+
+  public func configureGRPCHTTP2ServerTransportPipeline(
```

Comment:
We want a pipeline helper that we can use for NIOPosix _and_ NIOTS, which means it should be in the grpc http2 core module. It also means that the config should be common across the two. This also doesn't need to be public, it can use `@_spi(Package)` (until we can use `package` access).

The approach I've taken for the client is to have "structured" configuration rather than a giant config blob. The NIOPosix/NIOTS transports will each have their own configs which will compose the common bits and add their own bits of specific config on the side. See also https://github.com/grpc/grpc-swift/pull/1880/files